src/connector/src/parser/mod.rs

```diff
@@ -224,105 +241,107 @@ impl OpAction for OpActionUpdate {
 }
 
 impl SourceStreamChunkRowWriter<'_> {
-    #[expect(
-        clippy::disallowed_methods,
-        reason = "FIXME: why zip_eq_fast leads to compile error?"
-    )]
+    /// Set the meta data of the original message for this row writer.
+    ///
+    /// This should always be called except for tests.
+    fn with_meta(self, row_meta: MessageMeta) -> Self {
+        Self {
+            row_meta: Some(row_meta),
+            ..self
+        }
+    }
+}
+
+impl SourceStreamChunkRowWriter<'_> {
     fn do_action<A: OpAction>(
         &mut self,
         mut f: impl FnMut(&SourceColumnDesc) -> Result<A::Output>,
-    ) -> Result<WriteGuard> {
-        let mut modify_col = Vec::with_capacity(self.descs.len());
-        self.descs
-            .iter()
-            .zip_eq(self.builders.iter_mut())
-            .enumerate()
-            .try_for_each(|(idx, (desc, builder))| -> Result<()> {
-                if desc.is_meta() || desc.is_offset() {
-                    return Ok(());
-                }
-                let output = if desc.is_row_id() {
-                    A::DEFAULT_OUTPUT
-                } else {
-                    f(desc)?
+    ) -> Result<()> {
+        let mut f = |desc: &SourceColumnDesc| {
```

Comment:
The input parameter is also named `f`, I am confused which is which.

Reply:
Renamed wrapped `f` with `f_with_meta`.